options.go

```diff
@@ -183,6 +183,9 @@ type Options struct {
 
 	// For testing purposes only
 	ConnFactory ConnectorFactory `yaml:"-" json:"-"`
+
+	// Enable LightStep Meta Event Logging
+	MetaEventLogging bool `yaml:"meta_event_logging" json:"meta_event_logging"`
```

Comment:
Making this an option sounds like leaving it unset amounts to opting out, i.e., this looks like an opt-in flag. It's not treated as an opt-out flag, however, it's mutated which feels wrong for an options struct. Should this be moved into a non-options struct?

Reply:
Would it be more helpful to rename it? `MetaEventReportingEnabled`?

Reply:
I would say `MetaEventReportingDisabled`, that way the user gets the default on behavior via Go's struct initialization, and has to set `true` to opt-out of the feature.

But I wouldn't modify the value passed in by the user, have a field in the `tracerImpl` which is set to `!MetaEventReportingDisabled && DevModeEnabledBySatellite()`.

Reply:
Shouldn't the satellite disable this feature except in dev-mode? It seems easier for the user to get started if this is on-by-default-as-long-as-the-satellite-says.

Reply:
The satellite either sends `devMode: true` or nothing, there's no associated `devMode: false`. If we wanted to add that, the tracers could check for it.

Reply:
I mean to presume `devMode: false` unless `devMode: true` is set. Then a default of `MetaEventReportingDisabled: false` would allow dev-mode satellites to enable meta-reporting clients out of the box, while production satellites would automatically leave meta-reporting disabled.

Reply:
I pushed a change that does effectively this, if the devmode flag isn't explicitly there then it'll disable reporting.